pilot/pkg/xds/internalgen.go

```diff
@@ -153,3 +162,52 @@ func (sg *InternalGen) Generate(proxy *model.Proxy, push *model.PushContext, w *
 	}
 	return res
 }
+
+func (sg *InternalGen) registerWorkload(con *Connection) {
+	if sg.Store == nil || !con.node.Metadata.AutoRegister {
+		return
+	}
+	proxy := con.node
+	if len(proxy.IPAddresses) == 0 {
+		adsLog.Errorf("auto registration of %v failed: missing IP addresses", proxy.ID)
+	}
+	if len(proxy.Metadata.Namespace) == 0 {
+		adsLog.Errorf("auto registration of %v failed: missing namespace", proxy.ID)
+	}
+	we := &networking.WorkloadEntry{
+		// TODO add remaining fields
+		Address:        proxy.IPAddresses[0],
+		Labels:         proxy.Metadata.Labels,
+		ServiceAccount: proxy.Metadata.ServiceAccount,
+	}
+	cfg := model.Config{
+		ConfigMeta: model.ConfigMeta{
+			GroupVersionKind: gvk.WorkloadEntry,
+			Name:             getWorkloadEntryName(we.Address),
+			Namespace:        proxy.Metadata.Namespace,
+			Labels:           proxy.Metadata.Labels,
+		},
+		Spec: we,
+	}
+	if _, err := sg.Store.Create(cfg); err != nil {
+		adsLog.Errorf("failed to register workload %s/%s: %v", cfg.Name, cfg.Namespace, err)
+	} else {
+		adsLog.Infof("registered workload entry %s/%s", cfg.Name, cfg.Namespace)
+	}
+}
+
+func (sg *InternalGen) unregisterWorkload(con *Connection) {
+	if sg.Store == nil || !con.node.Metadata.AutoRegister || len(con.node.IPAddresses) == 0 {
```

Comment:
why would a workload not be auto-deleted if it doesn't have any associated IP addresses?